### PR TITLE
@sweir27 => "SOLD" button label 

### PR DESCRIPTION
**Pull request by damassi**
This PR fixes a few places where an artwork that is has been purchased does not show up as "SOLD" in its description. 

Fixes https://github.com/artsy/auctions/issues/404 
Related https://github.com/artsy/auctions/issues/405

**To Test:**
```sh
git checkout -b damassi-bugfix-buy-now-button-transition master
git pull https://github.com/damassi/force.git bugfix-buy-now-button-transition
```

**Additional Fixes:**
- Fixed all linter errors in `apps/auctions`
- Fixed a zero-state error in the slider, where result-sets containing no results would return a "Loading results" label

screen shot 2017-04-27 at 4 41 42 pm

- http://localhost:5000/auction/chris-buy-now-bug (look for francine-tint-untitled - third over)
---
screen shot 2017-04-27 at 5 34 25 pm

- http://localhost:5000/artwork/francine-tint-untitled (right-hand column)
---
screen shot 2017-04-27 at 5 32 54 pm

- http://localhost:5000/artwork/linda-carrara-ennui (search "francine" in artworks)
---
screen shot 2017-04-27 at 5 32 21 pm

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -2,5 +2,14 @@
...
-  ]
+  ],
+  "env": {
+    "browser": true,
+    "mocha": true,
+    "node": true
+  },
+  "rules": {
+    "camelcase": 0,
+    "no-new": 0
```
Comment (damassi) [inline .eslintrc]: *Our API responses and existing Backbone code

File: desktop/apps/auction/client/actions.js [hunk 2]
```diff
@@ -29,31 +29,31 @@ export const UPDATE_SORT = 'UPDATE_SORT'
 export const UPDATE_TOTAL = 'UPDATE_TOTAL'
 
 // Action creators
-export function decrementFollowedArtistsPage() {
+export function decrementFollowedArtistsPage () {
```
Comment (sweir27) [inline desktop/apps/auction/client/actions.js]: Thank you for fixing this!! (although side note, having the space looks so odd to me)

File: desktop/apps/auction/client/actions.js [hunk 3]
```diff
@@ -81,8 +81,8 @@ export function fetchArtworks() {
       }
 
       const aggregations = filter_sale_artworks.aggregations
-      const artistAggregation = aggregations.filter((agg) => agg.slice == 'ARTIST')
-      const mediumAggregation = aggregations.filter((agg) => agg.slice == 'MEDIUM')
+      const artistAggregation = aggregations.filter((agg) => agg.slice === 'ARTIST')
```
Comment (sweir27) [inline desktop/apps/auction/client/actions.js]: 😞

File: desktop/apps/auction/client/index.js [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
 import ConfirmRegistrationModal from '../../../components/credit_card/client/confirm_registration.coffee'
 import CurrentUser from '../../../models/current_user.coffee'
+import MyActiveBids from '../../../components/my_active_bids/view.coffee'
 import mediator from '../../../lib/mediator.coffee'
 import mountAuctionBlock from '../../../components/auction_block/index.jsx'
-import MyActiveBids from '../../../components/my_active_bids/view.coffee'
-import { setupCommercialFilter } from './commercial_filter'
 import { data as sd } from 'sharify'
-import _ from 'underscore'
+import { setupCommercialFilter } from './commercial_filter'
```
Comment (sweir27) [inline desktop/apps/auction/client/index.js]: What's your strategy for alphabetization? Is it based on the file path or the imported class/function?

**Comment by devangt** [comment]
👍 is this in staging? If so i can QA today.

File: desktop/apps/auction/components/basic_checkbox/index.js [hunk 5]
```diff
@@ -1,13 +1,13 @@
-import classNames from 'classnames'
+import PropTypes from 'prop-types'
 import React from 'react'
+import classNames from 'classnames'
```
Comment (sweir27) [inline desktop/apps/auction/components/basic_checkbox/index.js]: Like here, `c` comes before `P`? (I promise I was attempting to keep them organized haha).

File: desktop/apps/auction/components/filter_sort/index.js [hunk 6]
```diff
@@ -42,6 +44,12 @@ function FilterSort(props) {
   )
 }
 
+FilterSort.propTypes = {
```
Comment (sweir27) [inline desktop/apps/auction/components/filter_sort/index.js]: What are the chances we could start using Typescript and stop using prop types? (not in this PR obviously)

**Comment by sweir27** [comment]
@devangt not in staging yet! Will let you know when it is. 😄

File: desktop/apps/auction/client/actions.js (hunk 2 above)
Comment (damassi) [inline desktop/apps/auction/client/actions.js]: Yeah IMO its a controversial `standard` decision

File: desktop/apps/auction/client/index.js (hunk 4 above)
Comment (damassi) [inline desktop/apps/auction/client/index.js]: Natural sort (cmd+p sort:natural) -- in common usage its

```
*
$
Bar 
Foo
Hello
far
boo 
helloThere
{ Bar } 
{ Foo }
{ Hello }
{ bar }
{ foo }

Here's a good example with a lot of deps: https://github.com/artsy/prediction/blob/master/src/server.js
https://github.com/artsy/prediction/blob/master/src/routes/bidder/components/BidderLayout.js

File: desktop/apps/auction/components/basic_checkbox/index.js (hunk 5 above)
Comment (damassi) [inline desktop/apps/auction/components/basic_checkbox/index.js]: Natural sort takes into account cases, which often indicate different types; (`SomeClass` vs `someFunction`, for example.) 

Also! I know this is all super minor, I just tend to sort things whenever I come across them.

File: desktop/apps/auction/components/filter_sort/index.js (hunk 6 above)
Comment (damassi) [inline desktop/apps/auction/components/filter_sort/index.js]: I think thats where things are headed, but of course for a large discussion. `standard-react` screams without them, which in the long term is probably more of a good thing than bad as data dependencies change over time.

File: desktop/apps/auction/client/index.js (hunk 4 above)
Comment (sweir27) [inline desktop/apps/auction/client/index.js]: Thank you 🙌 !!!

File: desktop/apps/auction/components/basic_checkbox/index.js (hunk 5 above)
Comment (sweir27) [inline desktop/apps/auction/components/basic_checkbox/index.js]: 👍 no no it's good! I like when things are consistent, just wasn't sure how to deal with these cases (which is why my attempts were odd).

**Comment by katarinabatina** [comment]
could we make sold title case --> Sold vs SOLD

File: desktop/apps/auction/components/basic_checkbox/index.js (hunk 5 above)
Comment (damassi) [inline desktop/apps/auction/components/basic_checkbox/index.js]: Ah yeah, most editors come with different sort strategies behind the command panel

**Comment by damassi** [comment]
@katarinabatina - yup, will update.

**Pull request merged**